xbmc/windowing/gbm/drm/DRMAtomic.cpp

```diff
@@ -267,6 +249,32 @@ bool CDRMAtomic::AddProperty(CDRMObject* object, const char* name, uint64_t valu
   return true;
 }
 
+bool CDRMAtomic::AddPropertyValueByName(struct drm_object* object,
+                                        const char* name,
+                                        const char* type)
+{
+  for (uint32_t i = 0; i < object->props->count_props; i++)
+  {
+    if (!strcmp(object->props_info[i]->name, name))
+    {
+      for (uint32_t j = 0; j < object->props_info[i]->count_enums; j++)
+      {
+        if (!strcmp(object->props_info[i]->enums[j].name, type))
+        {
+          if (drmModeAtomicAddProperty(m_req, object->id, object->props_info[i]->prop_id,
+                                       m_gui_plane->props_info[i]->enums[j].value) < 0)
+          {
+            CLog::Log(LOGERROR, "CDRMAtomic::%s - could not add property %s", __FUNCTION__, name);
+            return false;
+          }
+        }
+      }
+    }
+  }
+  return true;
```

Comment:
You probably do not want to return `true` if the property or the enum value you are trying to set does not exist.